### PR TITLE
Add Single Node UI

### PR DESCRIPTION
**Pull request by tal-az**
TLDR It works

### CODE REVIEW

File: public/js/endpoints.js [hunk 1]
```diff
@@ -61,4 +61,124 @@ $(document).ready(function() {
...
+        if (!useDefaultKey) {
+            var privateKey = document.getElementById('privateKey').value;
+            data.privateKey = privateKey;
+        }
+        else {
+            data.privateKey = "default";
+        }
+    }
+
+    // Here, make your API call with the `data` object
```
Comment (GeneralSnuggles) [inline public/js/endpoints.js]: Okay, GPT

File: public/js/endpoints.js [hunk 2]
```diff
@@ -61,4 +61,124 @@ $(document).ready(function() {
...
+    console.log('Form data:', data);
```
Comment (GeneralSnuggles) [inline public/js/endpoints.js]: Is this logging still necessary?

File: public/js/endpoints.js [hunk 3]
```diff
@@ -61,4 +61,124 @@ $(document).ready(function() {
...
+        // handle install chipsec action
+        console.log('Installing CHIPSEC on selected endpoints:', selectedIds);
+    });
+
+    addNodeForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        const osType = document.getElementById('osType').value;
+        const hostId = document.getElementById('hostId').value;
+
+        console.log('Submitting form with data:', { osType, hostId }); // Add log here
```
Comment (GeneralSnuggles) [inline public/js/endpoints.js]: Is this logging still needed? also the comment

File: public/js/endpoints.js [hunk 4]
```diff
@@ -61,4 +61,124 @@ $(document).ready(function() {
...
+        else {
+            data.privateKey = "default";
+        }
+    }
+
+    // Here, make your API call with the `data` object
+    console.log('Form data:', data);
+
+    // Example API call (uncomment and modify as needed)
+    fetch('/api/agents/addNode/' + osType + "/" + hostId, {
```
Comment (GeneralSnuggles) [inline public/js/endpoints.js:139]: Why is this different than the AJAX endpoint? does AJAX call here?

**Review by GeneralSnuggles — COMMENTED** [review]
TLDR; remove some unnecessary(?) comments and logs and a bit of API clarification

**Comment by tal-az** [comment]
Note to self - add toas message utelizing this:
https://github.com/team408/BIOSQuery-Backend/blob/a477944a35a260c0d29d0c5731ea4ad32eb66579/public/js/endpoints.js

File: public/js/endpoints.js (hunk 4 above)
Comment (tal-az) [reply public/js/endpoints.js:139]: WDYM AJAX?

File: public/js/endpoints.js [hunk 5]
```diff
@@ -61,4 +61,116 @@ $(document).ready(function() {
...
+                // Handle success (e.g., refresh the list of endpoints)
+                location.reload();
+            },
+            error: function(error) {
+                // Handle error
+                console.error('Error adding host:', error);
+            }
+        });
+    });
+});
```
Comment (maziland) [inline public/js/endpoints.js]: Remove all these lines. Your already have an event listener for submit on line 139.
Also this ajax url is wrong (it has `.js` in the url)

**Review by maziland — CHANGES_REQUESTED** [review]
One very small change.
Would take 5 seconds to change :)

File: public/js/endpoints.js (hunk 5 above)
Comment (cayertall) [reply public/js/endpoints.js]: Got it, now i also understand Dar's resposne.
deleting and merging.

File: public/js/endpoints.js (hunk 4 above)
Comment (tal-az) [reply public/js/endpoints.js:139]: Got it after Mazi's response, resolved.

**Review by maziland — APPROVED** [review]
